src/components/content/registeredServices/tree/RegisteredServicesTree.tsx

```diff
@@ -16,15 +18,62 @@ export function RegisteredServicesTree({
     selectedKeyInTree: React.Key;
     setSelectedKeyInTree: (selectedKey: React.Key) => void;
 }): React.JSX.Element {
-    function isParentTreeSelected(selectedKeyInTree: React.Key): boolean {
+    const [searchValue, setSearchValue] = useState('');
+    const [filteredTreeData, setFilteredTreeData] = useState<DataNode[]>(treeData);
+
+    const filterTreeData = useCallback((data: DataNode[], search: string): DataNode[] => {
+        return data
+            .map((node) => {
+                if (node.children) {
+                    const filteredChildren = filterTreeData(node.children, search);
+                    if (filteredChildren.length > 0) {
+                        return { ...node, children: filteredChildren };
+                    }
+                }
+                if (node.title?.toString().toLowerCase().includes(search.toLowerCase())) {
+                    return node;
+                }
+                return null;
+            })
+            .filter((node): node is DataNode => node !== null);
+    }, []);
+
+    const findFirstLeafNodeKey = useCallback((data: DataNode[]): React.Key | null => {
+        for (const node of data) {
+            if (node.children && node.children.length > 0) {
+                const leafKey = findFirstLeafNodeKey(node.children);
+                if (leafKey) {
+                    return leafKey;
+                }
+            } else {
+                return node.key;
+            }
+        }
+        return null;
+    }, []);
+
+    useEffect(() => {
```

Comment:
we must avoid an effect here. 
Similar to the ServicTree... we just need to update filtered data based on `searchValue` state

Reply:
this must be still fixed.